api/share/requests/__init__.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC
+
+
+class RequestObject(ABC):
+    pass
+
+
+class InvalidRequestObject(RequestObject):
+
+    def __init__(self):
+        self.errors = []
+
+    def add_error(self, parameter, message):
+        self.errors.append({'parameter': parameter, 'message': message})
+
+    def has_errors(self):
+        return len(self.errors) > 0
+
+    def __bool__(self):
+        return False
+
+
+class ValidRequestObject(RequestObject):
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        raise NotImplementedError
+
+    def __bool__(self):
+        return True
```

Comment:
Is it really necessary? Any object is truthy. Even without this method the following will work:
```py
o = ValidRequestObject()
if o:
    do_something()
```
and `bool(o)` will also return `True`.

And, of course, even with this method `o is True` will return `False`.

Reply:
It's needed for corresponding to the InvalidRequestObject, where it returns False

Reply:
Well you can leave a corresponding comment then, because to an outside observer that will be strange. To me, though, it's strange that one class "knows" about its _sibling's_ internal structure. 

If you need to point out that in some subclass the `__bool__` method is handled differently, I'd suggest defining that `__bool__` method in `RequestObject` instead of `ValidRequestObject`. It will still look a bit strange to me, but at least it will be clear (perhaps a comment can be added in the parent class) that you did it to give a signal that you override it in one of the subclasses (and an IDE such as PyCharm will show where exactly).

Reply:
Ok, I got it.